specs/core/1_shard-data-chains.md

```diff
@@ -374,12 +384,17 @@ def process_shard_attestations(beacon_state: BeaconState, shard_state: ShardStat
             pubkeys.append(beacon_state.validators[validator_index].pubkey)
             process_delta(beacon_state, shard_state, validator_index, get_base_reward(beacon_state, validator_index))
             attestation_count += 1
+    # Exit early if no participants
```

Comment:
What is the rationale behind this?

Reply:
A block proposer might presumably not have any attestations from the previous slot to include. Need to handle that case explicitly.

This was never the case for phase 0 attestations, because there is no value in including an empty attestation. You can just have an empty list of attestations if there were actually not attestations to include. Here, though, you could easily have a case where there were not attestatiosn from the previous slot but a block still needs to be able to be built

Reply:
> This was never the case for phase 0 attestations

I believe phase 0 attestations can be empty, i.e. `attestation.aggregation_bits` can be all zero bits. My guess is we don't need special logic in phase 1, similar to how we don't have special logic in phase 0.

Reply:
What is the valid signature for no participants?
If there isn't one, then we need this special case

Reply:
> What is the valid signature for no participants?

It's the G2 point at infinity (i.e. `0 * g2` where `g2` is the `G2` generator). The reason is that the "empty" pubkey is the G1 point at infinity (i.e. `0 * g1` where `g1` is the `G1` generator) and `e(g1, sig) = e(0 * g1, H(m))` holds iff `sig` equals `0 * g2`.

> If there isn't one, then we need this special case

There is one (and it's unique!)—no need for a special case :)